### PR TITLE
[WIP] Distgen templating tool

### PR DESCRIPTION
**Pull request by rpitonak**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,46 +1,43 @@
...
+MAINTAINER {{ spec.maintainer }}
 
-ENV NAME=bind \
-    VERSION=0 \
-    RELEASE=1 \
-    ARCH=x86_64
+ENV NAME={{ spec.envvars.name }} \
+    VERSION={{ spec.envvars.version }} \
+    RELEASE={{ spec.envvars.release }} \
+    ARCH={{ spec.envvars.arch }}
```
Comment (bkabrda) [inline Dockerfile:12]: You should be able to use `config.os.arch` and so you wouldn't need to specify this in the spec file.

**Comment by bkabrda** [comment]
LGTM overall, but let's wait for some more distgen evolution before we step to detailed review and merging.

File: Makefile [hunk 2]
```diff
@@ -2,6 +2,13 @@
 
 TYPE='AUTHORITATIVE'
 
+DISTRO = fedora-26-x86_64
+VERSION = 9
+DG = /home/rpitonak/.local/bin/dg
```
Comment (phracek) [inline Makefile:7]: This is wrong:( I don't have a home directory called rpitonak. I would prefer to use `/usr/bin/dg` which is provided by distgen package from COPR.

**Comment by phracek** [comment]
The container is running but failing on some case which is not related to templating tool https://github.com/container-images/bind/issues/8

**Comment by phracek** [comment]
LGTM after Makefile modifications. Let's merge it. It works.

**Pull request merged**